src/lib/core/clock_lowres.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2022, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include <signal.h>
+#include <string.h>
+#include <say.h>
+#include <pthread.h>
+#include "clock.h"
+#include "clock_lowres.h"
+
+#if defined(__APPLE__) || !defined(NDEBUG)
+/**
+ * A thread that initialized this module. Only owner thread
+ * is allowed to handle SIGALRM and use methods from this module.
+ */
+pthread_t owner;
+
+/** Check if current thread is owner. */
+bool
+clock_lowres_thread_is_owner(void)
+{
+	return pthread_self() == owner;
+}
+#endif
+
+/** Old sigaction for SIGALRM. */
+static struct sigaction osa;
+
+/** Granularity of clock update. */
+static const struct timeval LOW_RES_GRANULARITY = {
+	.tv_sec = 0,
+	.tv_usec = 10 * 1e3,
+};
+
+/** Monotonic time, updated once every LOW_RES_GRANULARITY of time unit. */
+double low_res_monotonic_clock = 0.0;
+
+/** A tick of low_res_clock, SIGALRM handler. */
+static void
+clock_monotonic_lowres_tick(int signum)
+{
+#ifdef __APPLE__
+	/**
+	 * We cannot guarantee that the signal is delivered
+	 * only to owner thread on MacOS. So use this workaround.
+	 * https://github.com/tarantool/tarantool/issues/7206
+	 */
+	if (!clock_lowres_thread_is_owner()) {
+		pthread_kill(owner, signum);
+		return;
+	}
+#else
+	(void)signum;
+	assert(clock_lowres_thread_is_owner());
+#endif
+	low_res_monotonic_clock = clock_monotonic();
+}
+
+void
+clock_lowres_signal_init(void)
+{
+#if defined(__APPLE__) || !defined(NDEBUG)
+	owner = pthread_self();
+#endif
+	low_res_monotonic_clock = clock_monotonic();
+	struct sigaction sa;
+	memset(&sa, 0, sizeof(sa));
+	sa.sa_handler = clock_monotonic_lowres_tick;
+	sa.sa_flags = SA_RESTART;
+	if (sigaction(SIGALRM, &sa, &osa) == -1)
+		panic_syserror("cannot set low resolution clock timer signal");
+
+	struct itimerval timer;
+	timer.it_interval = LOW_RES_GRANULARITY;
+	timer.it_value = LOW_RES_GRANULARITY;
+	if (setitimer(ITIMER_REAL, &timer, NULL) == -1)
+		panic_syserror("cannot set low resolution clock timer");
+}
+
+void
+clock_lowres_signal_reset(void)
+{
+	struct itimerval timer;
+	memset(&timer, 0, sizeof(timer));
+	if (setitimer(ITIMER_REAL, &timer, NULL) == -1)
+		say_syserror("cannot reset low resolution clock timer");
+
+	if (sigaction(SIGALRM, &osa, 0) == -1)
```

Comment:
Why not use `SIG_DFL`?

Reply:
Isn't it better to set the value that was before initialization?

Reply:
It's supposed to reset the signal handler on fork so it should use `SIG_DFL`:

https://github.com/tarantool/tarantool/blob/59ce25dd4ca3d6b1e4ef757eff6f6f65e7341683/src/main.cc#L216-L218

Reply:
fixed